### ISSUE TITLE
Hitting ENTER Key for Filtering

### ISSUE THREAD

**Issue by Tangol** [villagedefrance/OpenCart-Overclocked#121]
I've noticed that hitting the **ENTER Key** to filter searches, doesn't always work.

For example, at **Categories**, when you type in a category name and hit **enter**, you get this message, then the page refreshes and the search returns the correct result.
![err1](https://user-images.githubusercontent.com/3336728/36080218-58beafd2-0f95-11e8-8cd1-9950a9b308c7.jpg)

The same applies for the **Manufacturer**, **Downloads**, **Reviews**, and **Palettes** search (same error, then correct result).

Also, for **Product Returns**, **Customers**, **Suppliers**, **Affiliates**, **Product Labels Report**, **Products Stock Report**, **Product Profit Margins Report** and **Robots Online Report** - the filtering using the ENTER key doesn't work.
You have to press the **Filter** button.

It works as it should only for **Products**, **Orders** (partially - the **Total** search doesn't work at all), **Supplier Products** and **Customer Countries Report**.

Tested on commit https://github.com/villagedefrance/OpenCart-Overclocked/commit/bab3a37ee827422e9a5a1718447f14fa6914ad35

**Comment by villagedefrance** [comment]
Using the ENTER key on a Touch enabled web page is likely to fail, especially if there is more than one input field. I agree that instinctively it would be nice to be able to strike enter to validate but I think there are browser limitations on this, possibly because there is no ENTER key on mobile devices and all modern browsers are now mobile friendly.
Maybe I am missing something ... a css definition I don't know about or else.
At this time I can't think of a way of fixing this ... if that can be fixed at all.

**Comment by Tangol** [comment]
I understand, and this isn't really an issue having to click the **Filter** button, but what about the **Delete/Uninstall** message, when you hit Enter at **Categories**, **Manufacturer**, **Downloads**, **Reviews**, and **Palettes**?

**Comment by villagedefrance** [comment]
Yes, I see it too ... I will try to do something to prevent the message from showing.

**Comment by villagedefrance** [comment]
Ok, bad news: Javascript functions KeyCode, Which and CharCode are now deprecated!
The current code is therefore not working. I have replaced KeyCode with Which (apparently works better across browsers) but still not perfect. I tried the "Event.key" workaround but still unsuccessful.
Ref: [https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode](url)
I'm thinking of removing the script on all pages.
Any suggestions?
